core/services/relay/evm/evm.go

```diff
@@ -738,11 +740,25 @@ func (r *Relayer) NewLLOProvider(ctx context.Context, rargs commontypes.RelayArg
 		r.lggr.Info("Benchmark mode enabled, using dummy transmitter. NOTE: THIS WILL NOT TRANSMIT ANYTHING")
 		transmitter = bm.NewTransmitter(r.lggr, fmt.Sprintf("%x", privKey.PublicKey))
 	} else {
-		clients := make(map[string]wsrpc.Client)
+		clients := make(map[string]grpc.Client)
 		for _, server := range lloCfg.GetServers() {
-			client, err2 := r.mercuryPool.Checkout(ctx, privKey, server.PubKey, server.URL)
-			if err2 != nil {
-				return nil, err2
+			var client grpc.Client
+			switch r.mercuryCfg.Transmitter().Protocol() {
+			case "grpc":
+				client = grpc.NewClient(grpc.ClientOpts{
+					Logger:        r.lggr,
+					ClientPrivKey: privKey.PrivateKey(),
+					ServerPubKey:  ed25519.PublicKey(server.PubKey),
+					ServerURL:     server.URL,
+				})
+			case "wsrpc":
+				wsrpcClient, checkoutErr := r.mercuryPool.Checkout(ctx, privKey, server.PubKey, server.URL)
+				if checkoutErr != nil {
+					return nil, checkoutErr
+				}
+				client = wsrpc.GRPCCompatibilityWrapper{Client: wsrpcClient}
+			default:
+				return nil, fmt.Errorf("unsupported protocol %q", r.mercuryCfg.Transmitter().Protocol())
```

Comment:
If you would prefer validation of the config itself, you could use a type that implements `UnmarshalText(text []byte) error` instead of plain string.